### DISCUSSION TITLE
Differences with other plaintext time-trackers

### DISCUSSION THREAD

**Discussion by the-solipsist** [jotaen/klog#185] [Something else]
I'm trying to evaluate the relative strengths and weaknesses of different plaintext time trackers, like Emacs/Ledger/[Hledger timeclock](https://hledger.org/hledger.html#timeclock-format) (especially with a helper script like [tt](https://github.com/dave-burke/tt)), [Watson](https://github.com/TailorDev/Watson), and [Timewarrior](https://timewarrior.net/).

It might be useful to have a small comparison, or even an explanation of why the existing tools fell short leading you to write klog.

**Comment by jotaen** [comment]
I’ve written two posts on my blog covering some of the background thoughts: [this one](https://www.jotaen.net/9zRPA/klog-time-tracking-plain-textfiles/) shortly after I published the project initially, and [this one](https://www.jotaen.net/9rRZQ/klog-retrospective/) a few months ago.

It’s not a systematic comparison, but I also didn’t look too deeply into other CLI-based tools. The “tldr” is that I wanted a pleasant and expressive data format primarily, that would allow me to view and edit my data in any editor, without needing to rely on specific tools, or being forced into specific workflows. The klog CLI tool is mostly for data evaluation, and for a few convenience operations. The most important thing is the transparent file format, that’s meaningful on its own.
